client/my-sites/theme/main.jsx

```diff
@@ -71,48 +67,17 @@ const ThemeSheet = React.createClass( {
 	},
 
 	getInitialState() {
-		return { selectedAction: null };
+		return {
+			showPreview: false,
+		};
 	},
 
 	componentDidMount() {
 		window.scroll( 0, 0 );
 	},
 
-	hideSiteSelectorModal() {
-		this.setState( { selectedAction: null } );
-	},
-
-	isActive() {
-		const { id, currentTheme } = this.props;
-		return currentTheme && currentTheme.id === id;
-	},
-
-	onPrimaryClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else if ( this.isActive() ) {
-			this.props.customize( this.props, this.props.selectedSite );
-		} else if ( this.props.price ) {
-			this.selectSiteAndDispatch( 'purchase' );
-		} else {
-			this.selectSiteAndDispatch( 'activate' );
-		}
-	},
-
-	onPreviewButtonClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else {
-			this.selectSiteAndDispatch( 'customize' );
-		}
-	},
-
-	selectSiteAndDispatch( action ) {
-		if ( this.props.selectedSite ) {
-			this.props[ action ]( this.props, this.props.selectedSite, 'showcase-sheet' );
-		} else {
-			this.setState( { selectedAction: action } );
-		}
+	getTheme() {
+		return pick( this.props, [ 'id', 'name', 'author', 'screenshot', 'price', 'description', 'descriptionLong', 'supportDocumentation', 'download', 'taxonomies', 'stylesheet', 'active', 'purchased' ] );
```

Comment:
> It would be great to have a common theme object. One obstacle to that is that different endpoints return different fields.

That's good info. Looking at it again, my concern is the dependencies (eg: `customize()` from `state/themes/actions`, `isPremium()`, `selectedTheme` prop from `ThemesSiteSelectorModal`) which accept a theme object. IMO we should probably either standardize the theme object for these things or have them accept simpler props (eg: just theme slug).

I may be perpetuating the issue here rather than simplifying it (as I had intended). When I refactor this I'll look at each of those dependencies and see if I can construct simple objects for each one with just the data they need; then I'll be able to determine what data the button needs and specify that explicitly in the PropTypes.